cmd/blob/acceptance_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright © 2022 Meroxa, Inc. and Miquido
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	as "github.com/miquido/conduit-connector-azure-storage"
+	asSource "github.com/miquido/conduit-connector-azure-storage/source"
+	helper "github.com/miquido/conduit-connector-azure-storage/test"
+	"go.uber.org/goleak"
+)
+
+type CustomConfigurableAcceptanceTestDriver struct {
+	sdk.ConfigurableAcceptanceTestDriver
+
+	containerClient *azblob.ContainerClient
+}
+
+func (d *CustomConfigurableAcceptanceTestDriver) GenerateRecord(t *testing.T) sdk.Record {
+	record := d.ConfigurableAcceptanceTestDriver.GenerateRecord(t)
+
+	// Override Key
+	record.Key = sdk.RawData(fmt.Sprintf("file-%d.txt", time.Now().UnixMicro()))
+
+	// Override CreatedAt
+	// azurite does not support milliseconds precision
+	record.CreatedAt = record.CreatedAt.Truncate(time.Second)
+
+	return record
+}
+
+func (d *CustomConfigurableAcceptanceTestDriver) WriteToSource(t *testing.T, records []sdk.Record) (results []sdk.Record) {
+	for _, record := range records {
+		_ = helper.CreateBlob(
+			d.containerClient,
+			string(record.Key.Bytes()),
+			"text/plain",
+			string(record.Payload.Bytes()),
+		)
+	}
+
+	// No destination connector, return wanted records
+	return records
+}
+
+func TestAcceptance(t *testing.T) {
+	sourceConfig := map[string]string{
+		asSource.ConfigKeyConnectionString: helper.GetConnectionString(),
+		asSource.ConfigKeyContainerName:    "acceptance-tests",
+	}
+
+	testDriver := CustomConfigurableAcceptanceTestDriver{
+		ConfigurableAcceptanceTestDriver: sdk.ConfigurableAcceptanceTestDriver{
+			Config: sdk.ConfigurableAcceptanceTestDriverConfig{
+				Connector: sdk.Connector{
+					NewSpecification: as.Specification,
+					NewSource:        asSource.NewSource,
+					NewDestination:   nil,
+				},
+
+				SourceConfig:     sourceConfig,
+				GenerateDataType: sdk.GenerateRawData,
+
+				GoleakOptions: []goleak.Option{
+					// Routines created by Azure client
+					goleak.IgnoreTopFunction("internal/poll.runtime_pollWait"),
+					goleak.IgnoreTopFunction("net/http.(*persistConn).writeLoop"),
+				},
+
+				Skip: []string{
+					// When restarting snapshot position, bucket is read from the beginning
```

Comment:
IIRC, only the object metadata is read from the beginning, but the object contents, are not?

Reply:
Object's contents is also read while iterating over storage.

Reply:
And from what we know so far, it's not possible to avoid that? It could be come a bottleneck.

Reply:
Object's contents is downloaded separately:
https://github.com/miquido/conduit-connector-azure-storage/blob/main/source/iterator/snapshot.go#L118-L123

So you can potentially adjust the logic 🙂

Reply:
Thanks, that's a useful reference!